### PR TITLE
INTYGFV-15582: Refactored to/from-sickLeaveLength into SickLeaveLengt…

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**
…hInterval. Updated affected tests and filters.

### CODE REVIEW

File: web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java [hunk 1]
```diff
@@ -64,48 +65,39 @@ class FilterOnSickLeaveLength {
...
-        void shouldNotFilterOnSickLeaveLengthIfFromAndToIsNull() {
-            final var expectedSickLeave = createSjukFallEnhet(DIAGNOSIS_CODE, 12, PATIENT_ID);
-            final var anotherSickLeave = createSjukFallEnhet(DIAGNOSIS_CODE, 5, PATIENT_ID);
-            final var expectedSickLeaveList = List.of(expectedSickLeave, anotherSickLeave);
+            final var anotherSickLeave = createSjukFallEnhet(DIAGNOSIS_CODE, 42, PATIENT_ID);
             final var sickLeaves = List.of(expectedSickLeave, anotherSickLeave);
 
-            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves, null, null, Collections.emptyList(), null, null);
+            final var actualSickLeaveList = filterSickLeaves.filter(sickLeaves,
+                List.of(new SickLeaveLengthInterval(6, 12), new SickLeaveLengthInterval(12, 40), new SickLeaveLengthInterval(50, 100)),
```
Comment (mhornfeldt) [inline web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java:74]: We should add tests to see if filtering is correct when there is either from or to as null

File: web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImpl.java [hunk 2]
```diff
@@ -40,20 +41,21 @@ public FilterSickLeavesImpl(DiagnosisChapterService diagnosisChapterService, Cal
...
 
-    private boolean filterOnSickLeaveLength(SjukfallEnhet sickLeave, Integer fromSickLeaveLength, Integer toSickLeaveLength) {
-        if (fromSickLeaveLength == null || toSickLeaveLength == null) {
+    private boolean filterOnSickLeaveLengthIntervals(SjukfallEnhet sickLeave, List<SickLeaveLengthInterval> sickLeaveLengthIntervals) {
+        if (sickLeaveLengthIntervals == null || sickLeaveLengthIntervals.isEmpty()) {
             return true;
         }
-        return sickLeave.getDagar() >= fromSickLeaveLength && sickLeave.getDagar() <= toSickLeaveLength;
+        return sickLeaveLengthIntervals.stream()
+            .anyMatch(interval -> interval.getFrom() <= sickLeave.getDagar() && interval.getTo() >= sickLeave.getDagar());
```
Comment (mhornfeldt) [inline web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImpl.java:58]: If from or to is null then the filter should allow everything over / under that date. Is that how it behaves now?

File: web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java (hunk 1 above)
Comment (ChrilleSigerhed) [reply web/src/test/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImplTest.java:74]: My bad, i was not aware of that from/to could be null. Updated!

File: web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImpl.java (hunk 2 above)
Comment (ChrilleSigerhed) [reply web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/FilterSickLeavesImpl.java]: My bad, i was not aware of that from/to could be null. Updated!

**Review by mhornfeldt — APPROVED** [review]
